### PR TITLE
[feature] use webp for thumbnails

### PR DESCRIPTION
**Pull request by NyaaaWhatsUpDoc**
# Description

Updates our media processor to use webp for generating thumbnails instead of jpeg. This allows us to maintain transparency layers while generating thumbnails of similar quality that work out *a little bit* smaller in practice.

## Checklist

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have not leveraged AI to create the proposed changes.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [ ] I/we have made any necessary changes to documentation.
- [ ] I/we have added tests that cover new code.
- [ ] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

**Comment by NyaaaWhatsUpDoc** [comment]
tests still need updating

File: internal/media/ffmpeg.go [hunk 1]
```diff
@@ -64,23 +75,48 @@ func ffmpegClearMetadata(ctx context.Context, filepath string, ext string) error
...
+			"scale="+scale+","+
+
+			// YUVA 4:2:0 colorspace
+			"format=yuva420p",
+
+		// Only one frame
 		"-frames:v", "1",
+
+		// ~40% webp quality
+		"-qscale:v", "40",
```
Comment (tsmethurst) [inline internal/media/ffmpeg.go:114]: I think you need to use `-quality` for the webp codec, see https://ffmpeg.org/ffmpeg-codecs.html#Options-36

File: internal/media/ffmpeg.go [hunk 2]
```diff
@@ -100,10 +136,21 @@ func ffmpegGenerateStatic(ctx context.Context, filepath string) (string, error)
 	// Generate static with ffmpeg.
 	if err := ffmpeg(ctx, dirpath,
 		"-loglevel", "error",
+
```
Comment (tsmethurst) [inline internal/media/ffmpeg.go:139]: I think we can now just drop this `ffmpegGenerateStatic` function and just call `ffmpegGenerateThumb` for everything instead :)

File: internal/media/ffmpeg.go (hunk 1 above)
Comment (NyaaaWhatsUpDoc) [reply internal/media/ffmpeg.go:114]: qscale is the global way to set quality regardless of codec :)

Comment (NyaaaWhatsUpDoc) [reply internal/media/ffmpeg.go:114]: so it passes it to the appropriate quality parameter of the codec

File: internal/media/ffmpeg.go (hunk 2 above)
Comment (NyaaaWhatsUpDoc) [reply internal/media/ffmpeg.go:139]: ooh good point. that'll require more tests updating, you good with that?

Comment (NyaaaWhatsUpDoc) [reply internal/media/ffmpeg.go:139]: though wait actually, at the moment for static emoji we're generating full quality (lossless) forms of them, but then we'd be making a significant quality drop from png -> webm 40% quality

File: internal/media/ffmpeg.go (hunk 1 above)
Comment (tsmethurst) [reply internal/media/ffmpeg.go:120]: Ah okay, are you sure it works for web piss too?

File: internal/media/ffmpeg.go (hunk 2 above)
Comment (tsmethurst) [reply internal/media/ffmpeg.go:145]: Mm right. Maybe make this do `-lossless` webp then? Or web pee with default quality or something?

Comment (tsmethurst) [reply internal/media/ffmpeg.go:145]: And yes no problem re: tests :)

Comment (NyaaaWhatsUpDoc) [reply internal/media/ffmpeg.go:145]: I'm thinking maybe leave this one for another PR. still need to check what webp lossless will turn out like vs. png

File: internal/media/ffmpeg.go (hunk 1 above)
Comment (NyaaaWhatsUpDoc) [reply internal/media/ffmpeg.go:120]: yeee, I tried both ways and both files had the same sha256 sum :)

File: internal/media/ffmpeg.go (hunk 2 above)
Comment (tsmethurst) [reply internal/media/ffmpeg.go:145]: Alright no probs, I'll just update the non-emoji media for now

**Comment by tsmethurst** [comment]
Closes https://github.com/superseriousbusiness/gotosocial/issues/3041

**Pull request merged**